modules/programs/rofi-power.nix

```diff
@@ -0,0 +1,112 @@
+{ config, lib, pkgs, ... }:
+
+# Largely based on adnan360's power.sh gist:
+# https://gist.github.com/adnan360/f86012baeb4c9ca4f1af033550b03033
```

Comment:
Since the code is mostly copied in here I think you'd need to verify that the code in the gist is licensed in a way that is compatible with MIT.

Also, I'm generally not fond of the idea of maintaining packages within HM. Ideally they should go into Nixpkgs. Effectively, the script could be cleaned up and generalized a bit, packaged in Nixpkgs, then referred to here. Then it would be useful to all Nix users, not just HM ones.

Reply:
Honestly, I'm thinking I'm going to strip most of this out into a package, with a much smaller module to configure it, though I am definitely going to take your suggestions. I'll probably leave both of them in my soon-to-publish `nur-packages` repo unless/until somebody decides it merits being in Nixpkgs. Thanks for all the suggestions.

Reply:
(Also, _Blood and Ink_ I was so new to Nix as a language when I did this. This code was a mess.)